apps/ledger-live-mobile/src/screens/Portfolio/TabSection.tsx

```diff
@@ -11,130 +15,286 @@ import {
 
+export default function TabSection({
+  handleToggle,
+  handleAssetsContentSizeChange,
+  handleAccountsContentSizeChange,
+  handleButtonLayout,
+  onPressButton,
+  initialTab,
+  showAssets,
+  assetsLength,
+  accountsLength,
+  assetsFullHeight,
+  accountsFullHeight,
+  maxItemsToDysplay,
+}: TabSectionProps) {
+  const { t } = useTranslation();
+  const [hasAnimated, setHasAnimated] = useState(false);
+  const [displayHeight, setDisplayHeight] = useState(
+    showAssets ? assetsFullHeight : accountsFullHeight,
+  );
+  const hasInitializedAssets = useRef(false);
+  const hasInitializedAccounts = useRef(false);
+  const prevAssetsLength = useRef(assetsLength);
+  const prevAccountsLength = useRef(accountsLength);
+
+  useEffect(() => setHasAnimated(true), []);
+
+  useEffect(() => {
+    resetInitializationOnDataChange(
+      assetsLength,
+      accountsLength,
+      prevAssetsLength,
+      prevAccountsLength,
+      hasInitializedAssets,
+      hasInitializedAccounts,
+    );
+  }, [assetsLength, accountsLength]);
+
+  useEffect(() => {
+    if (!hasInitializedAssets.current && showAssets) {
+      return handleFirstRenderFrameAnimation(
+        hasInitializedAssets,
+        setDisplayHeight,
+        assetsFullHeight,
+      );
+    }
+  }, [showAssets, assetsFullHeight]);
+
+  useEffect(() => {
+    if (!hasInitializedAccounts.current && !showAssets) {
+      return handleFirstRenderFrameAnimation(
+        hasInitializedAccounts,
+        setDisplayHeight,
+        accountsFullHeight,
+      );
+    }
+  }, [showAssets, accountsFullHeight]);
+
+  useEffect(() => {
+    return handleTabChangeHeight(
+      showAssets,
+      assetsFullHeight,
+      accountsFullHeight,
+      setDisplayHeight,
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [showAssets]); // IMPORTANT: Only on tab change, not on height changes
+  return (
+    <>
+      <Box height={40} mb={16}>
+        <TabSelector
+          key={initialTab} // Force remount when initialTab changes to sync internal state
+          labels={[
+            { id: TAB_OPTIONS.Assets, value: t("assets.title") },
+            { id: TAB_OPTIONS.Accounts, value: t("accounts.title") },
+          ]}
+          onToggle={handleToggle}
+          initialTab={initialTab}
+        />
+      </Box>
+      <Box testID="portfolio-assets-layout" minHeight={displayHeight}>
+        {showAssets ? (
+          <Animated.View
+            key="assets-view"
+            entering={hasAnimated ? SlideInLeft.duration(ANIMATION_DURATION_MS) : undefined}
+            exiting={hasAnimated ? SlideOutLeft.duration(ANIMATION_DURATION_MS) : undefined}
+            style={{ height: assetsFullHeight }}
+          >
+            <AssetsListView
+              sourceScreenName={ScreenName.Portfolio}
+              limitNumberOfAssets={maxItemsToDysplay}
+              onContentChange={handleAssetsContentSizeChange}
+            />
+            {assetsLength >= maxItemsToDysplay && (
```

Comment:
(minor typo)
```suggestion
            {assetsLength >= maxItemsToDisplay && (
```

also question: is React Native fine with rendering `{false}` ? i remember in the past it was better to do ternary but maybe it's only issue for `string` and `""` cases ?